### ISSUE TITLE
Setup CI pipeline and static analysis tools

### ISSUE THREAD

**Issue by parthopdas** [parthopdas/tddstud10#4]
- FxCop, 
- StyleCop, 
- FSharpLint 
- AppVeyor

**Comment by Henrik-R** [comment]
@parthopas I would like to help out with this issue, if you are interested

**Comment by parthopdas** [comment]
@Henrik-R Absolutely!

Couple of points:
1. I have already done the appveyor, you may want to add yourself to the build notifications to do that download this file, change extension to yml, add your email and commit it at the root. 
[appveyor.yml.txt](https://github.com/parthopdas/tddstud10/files/375919/appveyor.yml.txt)
2. Once you enable FxCop, StyleCop and FSharpLint - you will get a bunch of errors - to start off you may disable all of them - eventually once the infra is in place we will enable the warnings back one by one.
3. Please also change settings to covert build warnings to errors
4. As a stretch please see if you can convert the project to use paket completely (it will make the dependency management much easier)
5. Also as a stretch, please add an option to build.cmd to do debug or release builds, default being release

 Thanks for jumping in! And if you need anything additional info, please feel free to ping me directly at this@parthopdas.com

**Comment by HenrikRxn** [comment]
Added appveyor.yml in #19

**Comment by HenrikRxn** [comment]
@parthopdas I suggest closing this issue and opening 2 or more separate ones for the following reasons

- I am not going to pursue FSharpLint as it does not work out of the box and I know nothing about F# developement
- Smaller chunks of work make it easier for other people to jump in
- You avoid issues that are open for a long time and tend to get stuck

I would suggest issue for 

1, FSharpLint
2. Paket (I would be happy to continue on that)

**Comment by parthopdas** [comment]
Agreed. Could you please take care of that? viz. opening two separate
issues and picking up the paket one?

On Aug 1, 2016 4:08 PM, "Henrik Røn" <notifications@github.com> wrote:

@parthopdas <https://github.com/parthopdas> I suggest closing this issue
and opening 2 or more separate ones for the following reasons

   - I am not going to pursue FSharpLint as it does not work out of the box
   and I know nothing about F# developement
   - Smaller chunks of work make it easier for other people to jump in
   - You avoid issues that are open for a long time and tend to get stuck

I would suggest issue for

1, FSharpLint
2. Paket (I would be happy to continue on that)

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/parthopdas/tddstud10/issues/4#issuecomment-236496900>,
or mute the thread

**Comment by HenrikRxn** [comment]
Will do tonight (i.e. in 12 hours time)

**Comment by HenrikRxn** [comment]
Closing after moving FSharpLint and Paket to new issues
